modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/MuleDeployableProjectModelBuilder.java

```diff
@@ -0,0 +1,457 @@
+
+  public static final String CLASSLOADER_MODEL_JSON_DESCRIPTOR = "classloader-model.json";
+  public static final String CLASSLOADER_MODEL_JSON_PATCH_DESCRIPTOR = "classloader-model-patch.json";
+
+  public static final String CLASSLOADER_MODEL_JSON_DESCRIPTOR_LOCATION =
+      "META-INF/mule-artifact/" + CLASSLOADER_MODEL_JSON_DESCRIPTOR;
+  public static final String CLASSLOADER_MODEL_JSON_PATCH_DESCRIPTOR_LOCATION =
+      "META-INF/mule-artifact/" + CLASSLOADER_MODEL_JSON_PATCH_DESCRIPTOR;
+
+  public static final String CLASS_LOADER_MODEL_VERSION_120 = "1.2.0";
+  public static final String CLASS_LOADER_MODEL_VERSION_110 = "1.1.0";
+
+  private static final String GROUP_ID = "groupId";
+  private static final String ARTIFACT_ID = "artifactId";
+
+  private final File projectFolder;
+
+  public MuleDeployableProjectModelBuilder(File projectFolder) {
+    this.projectFolder = projectFolder;
+  }
+
+  @Override
+  public DeployableProjectModel build() {
+    AppClassLoaderModel packagerClassLoaderModel = getAppPackagerClassLoaderModel(getClassLoaderModelDescriptor(projectFolder));
+
+    Map<ArtifactCoordinates, BundleDescriptor> bundleDescriptors = packagerClassLoaderModel.getDependencies().stream()
+        .collect(toMap(Artifact::getArtifactCoordinates, artifact -> buildBundleDescriptor(artifact.getArtifactCoordinates())));
+
+    Map<BundleDescriptor, List<BundleDependency>> additionalPluginDependencies =
+        getAdditionalPluginDependencies(packagerClassLoaderModel, bundleDescriptors);
+
+    List<BundleDependency> dependencies = getDependencies(packagerClassLoaderModel);
+
+    Set<BundleDescriptor> sharedLibraries = getSharedLibraries(packagerClassLoaderModel, bundleDescriptors);
+
+    List<String> packages =
+        packagerClassLoaderModel.getPackages() != null ? Arrays.asList(packagerClassLoaderModel.getPackages()) : emptyList();
+    List<String> resources =
+        packagerClassLoaderModel.getResources() != null ? Arrays.asList(packagerClassLoaderModel.getResources()) : emptyList();
+
+    return new DeployableProjectModel(packages,
+                                      resources,
+                                      buildBundleDescriptor(packagerClassLoaderModel.getArtifactCoordinates()),
+                                      getModelResolver(packagerClassLoaderModel.getArtifactCoordinates()),
+                                      projectFolder, dependencies, sharedLibraries,
+                                      additionalPluginDependencies);
+  }
+
+  private Map<BundleDescriptor, List<BundleDependency>> getAdditionalPluginDependencies(AppClassLoaderModel packagerClassLoaderModel,
+                                                                                        Map<ArtifactCoordinates, BundleDescriptor> bundleDescriptors) {
+    return packagerClassLoaderModel
+        .getAdditionalPluginDependencies().map(
```

Comment:
can you reformat this to be more readable? You can play with the line breaks to influence the formatter